test/geometry_tests.cpp

```diff
@@ -10,156 +10,241 @@
 
+	// Check that the pixels are as expected
+	EXPECT_THAT(pixels, ElementsAreArray(expected));	
 }
 
-TEST(Point2DPlanning_Griz_Test, bresenham_conversion_octant1_test) {
+TEST(Point2DPlanning_Griz_Test, bresenham_conversion_going_down_and_left_test) {
+	// Start and end points
+	const Cell start {5,3};
+	const Cell end {0,0};
+
 	// Get the points for the line connecting (0,0) and (5,3)
-	auto v = bresenham_conversion(0, 0, 5, 3);
+	const auto pixels = bresenham_conversion(start, end);
 
 	// Check only one point was produced
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_EQ(pixels.size(), 6);
 
-	// Check the first point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 0, 0, v), true);	
+	// expected vector
+	const std::vector<Cell> expected {{5, 3}, {4, 2}, {3, 2}, {2, 1}, {1, 1}, {0, 0}};
 
-	// Check the last point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(1, 5, 3, v), true);	
+	// Check that the pixels are as expected
+	EXPECT_THAT(pixels, ElementsAreArray(expected));
+}
 
-	// Get the points for the line connecting (5,3) and (0,0)
-	v = bresenham_conversion(5, 3, 0, 0);
+TEST(Point2DPlanning_Griz_Test, raytrace_single_point_test) {
+	const Cell startPix = {0, 0};
+	const Cell endPix = {0, 0};
+	const int max_ray_length = 10; ///< This is an arbitrary number
+	const auto pixels = raytrace(startPix, endPix, max_ray_length);
 
-	// Check only one point was produced
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_EQ(pixels.size(), 1);	
+}
+
+TEST(Point2DPlanning_Griz_Test, raytrace_horz_line_going_right_test) {
+	const Cell start = {0, 0};
+	const Cell end = {5, 0};
+	const int max_length = 10; ///< This is an arbitrary number
+	const auto pixels = raytrace(start, end, max_length);	
+
+	// Expected vector
+	const std::vector<Cell> expected{{0, 0}, {1, 0}, {2, 0}, {3, 0}, {4, 0}, {5, 0}};
+
+	EXPECT_EQ(pixels.size(),(6));	
+	EXPECT_THAT(pixels,ElementsAreArray(expected));
+}
+
+TEST(Point2DPlanning_Griz_Test, raytrace_horz_line_going_left_test) {
+	const Cell start = {4, 5};
+	const Cell end = {-1, 5};
+	const int max_length = 10; ///< This is an arbitrary number
+	const auto pixels = raytrace(start, end, max_length);	
+
+	// Expected vector
+	const std::vector<Cell> expected{{4, 5}, {3, 5}, {2, 5}, {1, 5}, {0, 5}, {-1, 5}};
+
+	EXPECT_EQ(pixels.size(),(6));	
+	EXPECT_THAT(pixels,ElementsAreArray(expected));
+}
+
+TEST(Point2DPlanning_Griz_Test, raytrace_vert_line_going_up_test) {
+	const Cell start {-2, -3};
+	const Cell end {-2, 2};
+	const int max_length = 10; ///< This is an arbitrary number
+	const auto pixels = raytrace(start, end, max_length);
+
+	// Expected vector
+	const std::vector<Cell> expected{{-2, -3}, {-2, -2}, {-2, -1}, {-2, 0}, {-2, 1}, {-2, 2}};
+
+	EXPECT_EQ(pixels.size(),(6));	
+	EXPECT_THAT(pixels,ElementsAreArray(expected));
+}
+
+TEST(Point2DPlanning_Griz_Test, raytrace_vert_line_going_down_test) {
+	const Cell start = {1, 2};
+	const Cell end = {1,-3};
+	const int max_length = 10; ///< This is an arbitrary number
```

Comment:
`///<` is for doxygen. just still to `//` for internal one line comments

Reply:
Done